go/vt/vtctl/workflow/materializer.go

```diff
@@ -177,7 +216,6 @@ func (mz *materializer) generateBinlogSources(ctx context.Context, targetShard *
 			if !ok {
 				return nil, fmt.Errorf("unrecognized statement: %s", ts.SourceExpression)
 			}
-			filter := ts.SourceExpression
```

Comment:
Are we dropping the source expression completely? That seems wrong. I'd have expected us to still use this when it is not a multi-tenant migration and not a sharded migration.

Reply:
No, we are using it further below. The `filter` variable needed to be defined further down because of the change in logic.